oap-server/server-cluster-plugin/cluster-kubernetes-plugin/src/main/java/org/apache/skywalking/oap/server/cluster/plugin/kubernetes/NamespacedPodListInformer.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.skywalking.oap.server.cluster.plugin.kubernetes;
+
+import io.kubernetes.client.informer.SharedIndexInformer;
+import io.kubernetes.client.informer.SharedInformerFactory;
+import io.kubernetes.client.informer.cache.Lister;
+import io.kubernetes.client.openapi.ApiClient;
+import io.kubernetes.client.openapi.apis.CoreV1Api;
+import io.kubernetes.client.openapi.models.V1Pod;
+import io.kubernetes.client.openapi.models.V1PodList;
+import io.kubernetes.client.util.Config;
+import java.io.IOException;
+import java.util.List;
+import java.util.Optional;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public enum NamespacedPodListInformer {
+
+    /**
+     * contains remote collector instances
+     */
+    INFORMER;
+
+    private Lister<V1Pod> podLister;
+
+    public synchronized void init(ClusterModuleKubernetesConfig podConfig) {
+
+        try {
+            doStartPodInformer(podConfig);
+        } catch (IOException e) {
+            log.error("cannot connect with api server in kubernetes", e);
+        }
+    }
+
+    private void doStartPodInformer(ClusterModuleKubernetesConfig podConfig) throws IOException {
+
+        ApiClient client = Config.defaultClient();
+        CoreV1Api coreV1Api = new CoreV1Api(client);
+
+        SharedInformerFactory factory = new SharedInformerFactory();
```

Comment:
Pls, adopt to a custom thread pool to create daemon thread instead of the default cached thread pool. That's because the cluster coordination model is lack of `stop` phase, so we don't have any opportunity to invoke `stopAllRegisteredInformers`.

Reply:
add a hook when service shutdown ,thx for review again